jaxrs/src/main/java/io/airlift/jaxrs/AsyncResponseBinder.java

```diff
@@ -0,0 +1,110 @@
+package io.airlift.jaxrs;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import io.airlift.units.Duration;
+import jakarta.servlet.AsyncEvent;
+import jakarta.servlet.AsyncListener;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.ws.rs.container.AsyncResponse;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.Response;
+
+import java.util.Optional;
+import java.util.concurrent.Executor;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static io.airlift.jaxrs.AsyncResponseHandler.bindAsyncResponse;
+import static jakarta.ws.rs.core.Response.status;
+import static java.util.Objects.requireNonNull;
+
+public class AsyncResponseBinder
+{
+    private final HttpServletRequest servletRequest;
+    private boolean cancelOnClientDisconnect = true;
+    private Optional<Duration> timeout = Optional.empty();
+    private Function<Duration, Supplier<Response>> timeoutResponse = AsyncResponseBinder::defaultTimeoutResponse;
+
+    private AsyncResponseBinder(HttpServletRequest servletRequest)
```

Comment:
If we go with the @BeanParam solution, then can be:
```java
private AsyncResponseBinder(@Context HttpServletRequest servletRequest, @Suspended AsyncResponse asyncResponse)
```

Reply:
That won't work. You can inject @Suspended but it's not handled properly.

Reply:
Actually, it does but you need to annotate an argument with both @BeanParam and @Suspended